### PR TITLE
[WFCORE-5216] jboss-cli.sh script should use eval instead of exec

### PR DESCRIPTION
**Pull request by istudens**
https://issues.redhat.com/browse/WFCORE-5216

### CODE REVIEW

File: core-feature-pack/common/src/main/resources/content/bin/jboss-cli.sh [hunk 1]
```diff
@@ -82,8 +82,15 @@ JAVA_OPTS="$JAVA_OPTS -Dcom.ibm.jsse2.overrideDefaultTLS=true"
 
 LOG_CONF=`echo $JAVA_OPTS | grep "logging.configuration"`
 if [ "x$LOG_CONF" = "x" ]; then
-    exec "$JAVA" $JAVA_OPTS -Dlogging.configuration=file:"$JBOSS_HOME"/bin/jboss-cli-logging.properties -jar "$JBOSS_HOME"/jboss-modules.jar -mp "${JBOSS_MODULEPATH}" org.jboss.as.cli "$@"
+    eval \"$JAVA\" $JAVA_OPTS \
```
Comment (jfdenise) [inline core-feature-pack/common/src/main/resources/content/bin/jboss-cli.sh:85]: We are using exec due to the fix for https://issues.redhat.com/browse/JBEAP-8808.
We can't use eval there. At least in the general case.

**Review by jfdenise — CHANGES_REQUESTED** [review]
I don't think that the reported problem should be fixed in the jboss-cli script.sh. I asked for more information in the JIRA issue.

**Comment by DavyKoravand** [comment]
Hi, I would like to add that we also experienced jboss-cli problems due to exec being used after we updated WildFly from v10 to v21. We had issues using JAVA_OPTS parameters that used spaces, even though they were correctly encapsulated by quotes (we tried every escape/quote combination). We currently use a workaround where we make sure that JAVA_OPTS is empty when using the jboss-cli, but it would be nice to see eval being used again.

**Comment by jfdenise** [comment]
@DavyKoravand , could you provide some JAVA_OPTS values that create problems? Thank-you.

**Comment by jfdenise** [comment]
@istudens , instead of using eval to launch the CLI process, we should call the following just before to call exec:
JBOSS_MODULEPATH=$(eval "echo $JBOSS_MODULEPATH")
JAVA_OPTS=$(eval "echo $JAVA_OPTS")

**Comment by istudens** [comment]
@jfdenise ok, I will change the PR accordingly

**Comment by jfdenise** [comment]
@istudens , it happens that the problem reported with JAVA_OPTS will be fixed differently. Could you update your PR by just adding:
JBOSS_MODULEPATH=$(eval "echo $JBOSS_MODULEPATH")
and forget about the JAVA_OPTS.

**Comment by istudens** [comment]
I had to toss the testcase modification out as it wasn't working on Windows platform.

**Comment by jfdenise** [comment]
@istudens , could you make the addition to the test ignored on Windows and kept on other platforms?

**Comment by istudens** [comment]
I've finally get to the correct `eval` syntax which now even works with special characters in JBOSS_MODULEPATH like '(' etc. However, it is still `eval` which means it might be hacked to execute any arbitrary command if JBOSS_MODULEPATH is set by 
an evil user accordingly.

**Review by jfdenise — APPROVED** [review]
Thank-you, approved.

**Comment by jmesnil** [comment]
@istudens thanks

**Pull request merged**